elements/power.lua

```diff
@@ -160,58 +161,72 @@ local Update = function(self, event, unit)
 
 	power.disconnected = disconnected
 
-	local r, g, b, t
-
-	if(power.colorTapping and not UnitPlayerControlled(unit) and UnitIsTapDenied(unit)) then
-		t = self.colors.tapped
-	elseif(power.colorDisconnected and disconnected) then
-		t = self.colors.disconnected
-	elseif(displayType == ALTERNATE_POWER_INDEX and power.altPowerColor) then
-		t = power.altPowerColor
-	elseif(power.colorPower) then
-		local ptype, ptoken, altR, altG, altB = UnitPowerType(unit)
-
-		t = self.colors.power[ptoken]
-		if(not t) then
-			if(power.GetAlternativeColor) then
-				r, g, b = power:GetAlternativeColor(unit, ptype, ptoken, altR, altG, altB)
-			elseif(altR) then
-				-- As of 7.0.3, altR, altG, altB may be in 0-1 or 0-255 range.
-				if(altR > 1) or (altG > 1) or (altB > 1) then
-					r, g, b = altR / 255, altG / 255, altB / 255
+	local ptype, ptoken, altR, altG, altB = UnitPowerType(unit)
+	local t = self.colors.power[ptoken]
+	local r, g, b
+	local atlas = power.atlas or t.atlas
+	local bg = power.bg
+
+	if(power.useAtlas and atlas and displayType ~= ALTERNATE_POWER_INDEX) then
+		r, g, b = t[1], t[2], t[3]
+		power:SetStatusBarAtlas(atlas)
+		power:SetStatusBarColor(1, 1, 1)
```

Comment:
Actually I've tried not using atlas for the background and I use 0.5 for the multiplier. It was hard to see the power bar fill with some atlases. Because of that I would rather let the layout decide on the multiplier value since they could react if the contrast is not good enough.

I liked the atlas on the background actually, I could make it optional though.

I would rather not copy coloring code to the atlas section. If the layout prefers to use an atlas I suppose they like the default texture colors. They could also supply a custom atlas through `power.atlas`, alter `oUF.colors.power` or use PostUpdate if they need coloring for some reason.